### PR TITLE
Feat/fav and delete ideas

### PR DESCRIPTION
**Pull request by ivonne-hernandez**
What (if any) features are you implementing?
- Added functionality for favoriting and deleting ideas
What (if anything) did you refactor?
- Removed pseudocode 
Were there any issues that arose?
- Matching up the clicked card with the Idea instance was complicated
Is there anything that you need from your teammate?
- Please review the code and provide feedback as needed.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -47,15 +44,20 @@ function validateUserInput() {
...
+    } else {
+      starSource = "./assets/star.svg";
+    }
     ideaCardGrid.innerHTML +=
     `<div class="box-container" id="${ideas[i].id}">
       <div class="box-header-container">
-      <input class="star-btn" type="image" name="star button" src="./assets/star.svg" alt="picture-of-a-star">
-      <input class="delete-btn" type="image" name="delete button" src="./assets/delete.svg" alt="picture-of-an-x">
+        <input class="star-btn" type="image" name="star button" src="${starSource}" alt="picture-of-a-star">
+        <input class="delete-btn" type="image" name="delete button" src="./assets/delete.svg" alt="picture-of-an-x">
```
Comment (ivonne-hernandez) [inline main.js]: I created a starSource variable which assigns the appropriate image source depending on the star property of the idea.

File: main.js [hunk 2]
```diff
@@ -77,28 +79,28 @@ function deleteIdeaCard(ideaId) {
...
+  for (var i = 0; i < ideas.length; i++) {
+    if (ideaId === ideas[i].id) {
+      ideas[i].star = !ideas[i].star;
+      ideas[i].updateIdea();
+    }
+  }
 }
 
 function handleIdeaCardGridClick(event) {
+  var ideaId = Number(event.target.closest('.box-container').id);
```
Comment (ivonne-hernandez) [inline main.js:96]: I pulled this out of the if-statements because the ideaId will be used in both of the if-statements.

File: main.js [hunk 3]
```diff
@@ -77,28 +79,28 @@ function deleteIdeaCard(ideaId) {
...
-
-
-
-
-
-
 
+  if (event.target.classList.contains('star-btn')) {
+    favoriteIdeaCard(ideaId);
+  }
```
Comment (ivonne-hernandez) [inline main.js:103]: Invoke favoriteIdeaCard when the star-btn is clicked.

File: main.js [hunk 4]
```diff
@@ -77,28 +79,28 @@ function deleteIdeaCard(ideaId) {
...
 
-function favoriteIdeaCard() {
-
+function favoriteIdeaCard(ideaId) {
+  for (var i = 0; i < ideas.length; i++) {
+    if (ideaId === ideas[i].id) {
+      ideas[i].star = !ideas[i].star;
+      ideas[i].updateIdea();
+    }
+  }
```
Comment (ivonne-hernandez) [inline main.js:92]: The favoriteIdeaCard function iterates through the ideas array and flips the star property of the appropriate idea. It calls the updateIdea method of that instance.

File: Idea.js [hunk 5]
```diff
@@ -8,26 +6,30 @@ class Idea {
...
+  deleteFromStorage() {
+    var retrievedIdeas = localStorage.getItem('stringifiedIdeas');
+    var parsedIdeas = JSON.parse(retrievedIdeas);
+    for (var i = 0; i < parsedIdeas.length; i ++) {
+      if (this.id === parsedIdeas[i].id) {
+        parsedIdeas.splice(i, 1);
+      }
+    }
+    this.saveToStorage(parsedIdeas);
+  }
```
Comment (ivonne-hernandez) [inline Idea.js]: I just indented these lines.

File: Idea.js [hunk 6]
```diff
@@ -8,26 +6,30 @@ class Idea {
...
+  updateIdea() {
+    var retrievedIdeas = localStorage.getItem('stringifiedIdeas');
+    var parsedIdeas = JSON.parse(retrievedIdeas);
+    for (var i = 0; i < parsedIdeas.length; i ++) {
+      if (this.id === parsedIdeas[i].id) {
+        parsedIdeas[i].star = !parsedIdeas[i].star;
+      }
+    }
+    this.saveToStorage(parsedIdeas);
+  }
```
Comment (ivonne-hernandez) [inline Idea.js:34]: The updateIdea method retrieves the ideas array from local storage, flips the star value of the idea, and saves the updated array to local storage.

File: main.js [hunk 7]
```diff
@@ -77,28 +79,28 @@ function deleteIdeaCard(ideaId) {
...
-
-
-
 
+  if (event.target.classList.contains('star-btn')) {
+    favoriteIdeaCard(ideaId);
+  }
 
-//var starredIdeas = [display when user clicks on "Show Starred Ideas"];
+  displayCards();
```
Comment (ivonne-hernandez) [inline main.js:105]: Pulled this out because it should happen after either deleting or favoriting cards.

File: Idea.js (hunk 5 above)
Comment (reganlosey) [reply Idea.js]: Looks good, we'll need to refactor the rest of our indentation when remove all of our comments for better readability.

File: Idea.js (hunk 6 above)
Comment (reganlosey) [reply Idea.js:34]: Good idea. I figure when the Show Starred Ideas button gets built we can just use an if statement to evaluate the Boolean value of the star and if it returns true, we'll display it.

**Comment by edwardkrupicka** [comment]
I'm glad you got everything to work.

**Review by reganlosey — APPROVED** [review]
Looking good, functions the way it need to for this point in the iterations.

**Pull request merged**